pkg/scalers/azure_eventhub_scaler.go

```diff
@@ -116,15 +117,22 @@ func parseAzureEventHubMetadata(metadata, resolvedEnv map[string]string) (*Event
 }
 
 //GetUnprocessedEventCountInPartition gets number of unprocessed events in a given partition
-func (scaler *AzureEventHubScaler) GetUnprocessedEventCountInPartition(ctx context.Context, partitionID string) (newEventCount int64, err error) {
+func (scaler *AzureEventHubScaler) GetUnprocessedEventCountInPartition(ctx context.Context, partitionID string) (newEventCount int64, checkpoint Checkpoint, err error) {
 	partitionInfo, err := scaler.client.GetPartitionInformation(ctx, partitionID)
```

Comment:
This is the redundant call to GetPartitionInformation for Partition id in GetUnprocessedEventCountInPartition, we can pass the reference of partitionInfo eventhub.HubPartitionRuntimeInformation from IsActive and GetMetrics methods where this information has already been retrieved. Other than that this change will solve the issue of "Until the function is scaled to at least 1 manually, the keda-operator will never be able to get the checkpoint data"

Reply:
Ok thank you. I can change that. But now there are some merge conflicts. I will try to fix it.

Reply:
in isActive is no partitionInfo, only runtimeInfo but i can add it

Reply:
ok i'm done. I eliminate one call to GetPartitionInformation and fix the merge conflicts.